### PR TITLE
HIVE-27762: ObjectStore GetHelper do not need run jdo query if direct sql exception is unrecoverable

### PR DESCRIPTION
**Pull request by wecharyu**
### What changes were proposed in this pull request?
For the API in `ObjectStore` implemented both direct sql and jdo query, if the direct sql throws unrecoverable exception, we do not need fall back to jdo query.

### Why are the changes needed?
Unrecoverable exceptions will also be triggered in jdo query, we can fast fail it to improve performance.

### Does this PR introduce _any_ user-facing change?
No.

### Is the change a dependency upgrade?
No.

### How was this patch tested?
Add a unit test.

### CODE REVIEW

**Comment by wecharyu** [comment]
@deniskuzZ @zabetak @saihemanth-cloudera: could you help review this patch?

File: standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/RecoverableExceptions.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import java.sql.SQLIntegrityConstraintViolationException;
+import java.util.stream.Stream;
+
+public class RecoverableExceptions {
+  private static final Class<SQLException>[] unrecoverableSqlExceptions = new Class[]{
+      // TODO: collect more unrecoverable SQLExceptions
+      SQLIntegrityConstraintViolationException.class
+  };
+
+  public static boolean isRecoverableException(Throwable t) {
```
[the remark this answers is not in the document]
Comment (wecharyu) [reply standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/RecoverableExceptions.java]: Addressed.

**Comment by wecharyu** [comment]
Pipeline has passed. @saihemanth-cloudera

**Review by saihemanth-cloudera — APPROVED** [review]
LGTM +1. Need @deniskuzZ's approval also.

**Comment by wecharyu** [comment]
Gentle ping @deniskuzZ, please take a look at your convenience.

**Review by deniskuzZ — APPROVED** [review]
LGTM +1

**Pull request merged**